### PR TITLE
Fix STRIPE-967: Prevent dismissed banners from reappearing after tab navigation

### PR DESCRIPTION
**Pull request by kingsleyinfo**
## Summary

- Fixed stale `wc_stripe_settings_params` causing dismissed admin banner notices to reappear when navigating between the Settings and Payment Methods tabs
- Updated the shared `dismissNotice()` utility to sync the localized global params after a successful dismiss API call
- Switched the Stripe Tax banner from inline `apiFetch` to the shared `dismissNotice` utility for consistency

## Bug

STRIPE-967: UX issues with banners/notices in Stripe admin
https://linear.app/a8c/issue/STRIPE-967

## Changes

- `client/utils/index.js` — `dismissNotice()` now updates `wc_stripe_settings_params` with the dismissed state so the stale global doesn't cause banners to resurface after SPA tab navigation
- `client/settings/payment-settings/promotional-banner/stripe-tax-banner.js` — Replaced inline `apiFetch` call with the shared `dismissNotice` utility, so it also benefits from the global sync fix
- `client/utils/__tests__/dismiss-notice.test.js` — Added 5 regression tests covering the global param sync, edge cases (undefined global, null callback), and per-banner-type behavior

## Testing

- All 7 Jest tests pass (`npx jest --config tests/js/jest.config.js client/utils/__tests__/dismiss-notice.test.js`)
- To verify manually: dismiss any banner on the Payment Methods tab, switch to Settings tab and back — the banner should stay dismissed

## Scope Note

The bug report also mentions that banner dismissals are global (not per-user). This PR does not change that behavior — it only fixes the stale-data issue. The per-user scope question is a design decision for the owning team.

## Notes

This PR was created during the [Woo Bug Blitz](https://woohappyp2.wordpress.com/2026/03/18/extensions-bug-blitz/) via team Ceres' [`/bug-blitz` Claude skill](https://github.com/Automattic/public-resources-squad/blob/trunk/.claude/skills/bug-blitz/skill.md). The fix was authored with Claude Code assistance by a Happiness Engineer who encounters this bug in support tickets.

### CODE REVIEW

**Comment by wjrosa** [comment]
Hey @kingsleyinfo ! Is this PR ready for review?

**Comment by kingsleyinfo** [comment]
Hey @wjrosa, yes, this is ready for review.

File: changelog.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ xxxx-xx-xx - version 10.6.0
 * Fix - Better background color detection for block themes and allow fonts from fonts.bunny.net
 * Update - Shorten test mode messaging, add Test Mode badge on Blocks checkout, and add copy-to-clipboard for test card numbers
 * Fix - Prevent brief display of wrong title on classic checkout when Optimized Checkout is enabled
+* Fix - Prevent dismissed admin banner notices from reappearing after navigating between Settings and Payment Methods tabs
```
Comment (wjrosa) [inline changelog.txt:12]: The changelog entry for the readme.txt file is missing. Should be just like this one. It is also missing the milestone (see the GitHub Actions errors below)

**Review by wjrosa — APPROVED** [review]
Thanks for fixing this bug! Code is good and works as expected. It is just missing the changelog entry and the milestone.

**Review by diegocurbelo — APPROVED** [review]
Thanks for the fix @kingsleyinfo!.

The code changes look good, and it tests well

**Comment by daledupreez** [comment]
@coderabbitai resume

**Comment by malithsen** [comment]
Just a note that we have two types of banners. Ones that use `<Notice>` components and promotional banners that uses a different mechanism. This dismissal issue only seems to apply to those who use the former. Currently those are
 
- optimized-checkout-notice 
- display-order-customization-notice 
- optimized-checkout-first-method-notice

I am not able to replicate the issue on any of the promotional banners, as they live in the always-mounted `SettingsManager` parent, and their state is never re-derived on tab switches.

File: client/utils/index.js [hunk 2]
```diff
@@ -22,7 +22,45 @@ export const dismissNotice = ( noticeKey, callback ) => {
...
+						'show_bnpl_promotional_banner',
+					wc_stripe_show_oc_promotion_banner:
+						'show_oc_promotional_banner',
+					wc_stripe_show_stripe_tax_banner: 'show_stripe_tax_banner',
+					wc_stripe_show_customization_notice:
+						'show_customization_notice',
+					wc_stripe_show_optimized_checkout_notice:
+						'show_optimized_checkout_notice',
+					wc_stripe_show_stripe_first_method_notice:
+						'show_stripe_first_method_notice',
```
[the remark this answers is not in the document]
Comment (daledupreez) [reply client/utils/index.js:46]: Added in 4e4890373cf5464ac7c6a997d80f7d193315de43, which shifted to an `it.each()` structure to make it much easier to expand in future.

File: client/utils/__tests__/dismiss-notice.test.js [hunk 3]
```diff
@@ -32,4 +33,123 @@ describe( 'dismissNotice', () => {
...
+			);
+			expect( apiFetch ).toHaveBeenCalledWith( {
+				path: '/wc/v3/wc_stripe/settings/notice',
+				method: 'POST',
+				data: { [ noticeKey ]: 'no' },
+			} );
+
+			expect( callback ).toHaveBeenCalled();
+		}
+	);
```
[the remark this answers is not in the document]
Comment (daledupreez) [reply client/utils/__tests__/dismiss-notice.test.js:165]: Addressed in 34d06bb01de46b496bd3c70d904a0c63d1cb145d.
